public/app/features/manage-dashboards/utils/validation.ts

```diff
@@ -4,8 +4,29 @@ import { validationSrv } from '../services/ValidationSrv';
 
 export const validateDashboardJson = (json: string) => {
   try {
-    JSON.parse(json);
-    return true;
+    let dashboard = JSON.parse(json);
+    // If dashboard.tags provided && is Array
+    if (dashboard.tags && Array.isArray(dashboard.tags)) {
+      // Check dashboard.tags in Array for only Strings
+      let cont = true;
+      dashboard.tags.forEach((tag: string) => {
+        if (cont === true && typeof tag !== 'string') {
+          cont = false;
+        }
+      });
+      // Continue to import dashboard if Tags in correct format
+      if (cont) {
+        return true;
+      } else {
+        return 'error: tags expected Array of Strings';
+      }
+      // If dashboard.tags && not Array
+    } else if (dashboard.tags && !Array.isArray(dashboard.tags)) {
+      return 'error: tags expected Array';
+      // Else generic JSON test passes and no Tags provided, then import dashboard
+    } else {
+      return true;
+    }
   } catch (error) {
     return 'Not valid JSON';
   }
```

Comment:
```suggestion
  }
  if (dashboard.tags) {
    if (Array.isArray(dashboard.tags)) {
      const hasInvalidTag = dashboard.tags.some((tag: string) => typeof tag !== 'string');
      if (hasInvalidTag) {
        return 'error: tags expected Array of Strings';
      }
    } else {
      return 'error: tags expected Array';
    }
  }
      
```

Reply:
Having less `if else`s and variables makes it easier to read IMO. What do you think of those suggestions?